Refactor CMake with JRL CMake Modules v2

ahoarau opened the pull request:
⚠️DO NOT MERGE UNTIL https://github.com/jrl-umi3218/jrl-cmakemodules/pull/798 is merged ⚠️

This PR is a full rewrite of the CMake files with the [JRL CMake Modules v2](https://github.com/jrl-umi3218/jrl-cmakemodules/pull/798).

* Full rewrite of the CMake Files in modern CMake
* Remove submodules: archives can now be used
* Remove external libraries
* `c++17` minimum required
* `pixi` support
* 🚧 NIX CI is temporarily disabled
* Fix Documentation links
* Support for `find_package` components

```cmake
find_package(proxsuite 0.7.1 CONFIG REQUIRED COMPONENTS proxsuite vectorized)
```

ahoarau committed:
ignore more stuff

ahoarau committed:
ignore more stuff

ahoarau committed:
update gersemi conf
remove custom functions as they format in a strange way when favour-inlining (the default) is used.

ahoarau committed:
update gersemi conf
remove custom functions as they format in a strange way when favour-inlining (the default) is used.

ahoarau committed:
add KMP_DUPLICATE_LIB_OK directly in the __init__.py

ahoarau committed:
add KMP_DUPLICATE_LIB_OK directly in the __init__.py

ahoarau committed:
use raw string format in QPFunction to prevent warning during stub gen

ahoarau committed:
fix documentation compile option

ahoarau committed:
fix documentation compile option

ahoarau committed:
add pixi support

ahoarau committed:
add pixi support

ahoarau committed:
remove submodules

ahoarau committed:
remove submodules

ahoarau committed:
remove unused cmake-external

ahoarau committed:
remove unused cmake-external

ahoarau committed:
remove external libraries
and submodule dirs

ahoarau committed:
remove external libraries
and submodule dirs

ahoarau committed:
update doc

ahoarau committed:
update doc

ahoarau committed:
use cmake provided doxygen functions

ahoarau committed:
use cmake provided doxygen functions

ahoarau committed:
add white bg logo for documentation and readme

ahoarau committed:
add white bg logo for documentation and readme

ahoarau committed:
factorize.hpp: fix conversion warning
seen with sarah

ahoarau committed:
factorize.hpp: fix conversion warning
seen with sarah

ahoarau committed:
remove no longer needed condition in c++17

ahoarau committed:
remove no longer needed condition in c++17

ahoarau committed:
remove extra Eigen?Core header

ahoarau committed:
remove extra Eigen?Core header

ahoarau committed:
workspace.hpp: fix local variable shadowing a struct member

ahoarau committed:
workspace.hpp: fix local variable shadowing a struct member

ahoarau committed:
random_qp_problems.hpp: use static variables intead of global variables
fixes warnings in the proxsuite-test-utils

ahoarau committed:
random_qp_problems.hpp: use static variables intead of global variables
fixes warnings in the proxsuite-test-utils

ahoarau committed:
separate tests in cpp and python folders

ahoarau committed:
separate tests in cpp and python folders

ahoarau committed:
Refactor CMake with JRL CMake Modules v2

ahoarau committed:
Refactor CMake with JRL CMake Modules v2

ahoarau committed:
intregrate get-jrl-cmakemodules.cmake

ahoarau committed:
intregrate get-jrl-cmakemodules.cmake

ahoarau committed:
update jrl url to point to fork

ahoarau committed:
update jrl url to point to fork

ahoarau committed:
require jrl-cmakemodules version 2.0.0

ahoarau committed:
require jrl-cmakemodules version 2.0.0

ahoarau committed:
include stubs pyi files in the installation

ahoarau committed:
include stubs pyi files in the installation

ahoarau committed:
ros-ci: add verbosity

ahoarau committed:
ros-ci: add verbosity

ahoarau committed:
changelog: add jrl v2 entry

ahoarau committed:
changelog: add jrl v2 entry

ahoarau committed:
get-jrl.cmake: update with latest version
check dir and status msgs

ahoarau committed:
get-jrl.cmake: update with latest version
check dir and status msgs

ahoarau committed:
ci: minor cosmetics changes

ahoarau committed:
ci: minor cosmetics changes

ahoarau committed:
main cmake: add back copyright

ahoarau committed:
main cmake: add back copyright

ahoarau committed:
ci: add initial pixi ci

ahoarau committed:
ci: add initial pixi ci

ahoarau committed:
ci: update deps on arch ci

ahoarau committed:
ci: update deps on arch ci

ahoarau committed:
ci: use pixi to generate docs

ahoarau committed:
ci: use pixi to generate docs

ahoarau committed:
ci: add cache to pixi

ahoarau committed:
ci: add cache to pixi

ahoarau committed:
ci: condition pixi ci like the other cis

ahoarau committed:
ci: condition pixi ci like the other cis

ahoarau committed:
ci: add catch2 dep (arch)

ahoarau committed:
ci: add catch2 dep (arch)

ahoarau committed:
ci: add catch2 dep in conda

ahoarau committed:
ci: add catch2 dep in conda

ahoarau committed:
ci: add ceral and nanobind to conda env

ahoarau committed:
ci: add ceral and nanobind to conda env

ahoarau committed:
cmake: stubs depends on all targets built

ahoarau committed:
cmake: stubs depends on all targets built

ahoarau committed:
cmake: check vectorization support and add debug message

ahoarau committed:
cmake: check vectorization support and add debug message

ahoarau committed:
ros: add cereal catch2 deps

ahoarau committed:
ros: add cereal catch2 deps

ahoarau committed:
ros: add graphviz dep

ahoarau committed:
ros: add graphviz dep

ahoarau committed:
ci: enable caching for ros ci

ahoarau committed:
ci: enable caching for ros ci

ahoarau committed:
cmake: condition cereal dep

ahoarau committed:
cmake: condition cereal dep

ahoarau committed:
cmake: use fetchcontent as fallback for catch2
ubuntu 22 ships catch2 v2.x, which brings incompatible headers

ahoarau committed:
cmake: use fetchcontent as fallback for catch2
ubuntu 22 ships catch2 v2.x, which brings incompatible headers

ahoarau committed:
cmake: SYSTEM is not recognized on cmake 3.22

ahoarau committed:
cmake: SYSTEM is not recognized on cmake 3.22

ahoarau committed:
dense_maros_meszaros: include header locally

ahoarau committed:
dense_maros_meszaros: include header locally

ahoarau committed:
ci: add linux ci via APT

ahoarau committed:
ci: add linux ci via APT

ahoarau committed:
cmake: disable maros meszaros for now they crash

ahoarau committed:
cmake: disable maros meszaros for now they crash

ahoarau committed:
ci: add cereal on arch

ahoarau committed:
ci: add cereal on arch

ahoarau committed:
ci: add nanobind dep to arch

ahoarau committed:
ci: add nanobind dep to arch

ahoarau committed:
ci: add graphviz dep to arch

ahoarau committed:
ci: add graphviz dep to arch

ahoarau committed:
ci: add doxygen dep to arch

ahoarau committed:
ci: add doxygen dep to arch

ahoarau committed:
cmake: fallback to fetchcontent for cereal
cereal is not available on cmeel

ahoarau committed:
cmake: fallback to fetchcontent for cereal
cereal is not available on cmeel

ahoarau committed:
cmake : remove version min for cereal

ahoarau committed:
cmake : remove version min for cereal

ahoarau committed:
pyproject: require nanobind >=2.5.0

ahoarau committed:
pyproject: no min for nanobind

ahoarau committed:
pyproject: nanobind is not compatible with python <3.9

ahoarau committed:
cmake: fallback to fetchcontent for nanobind

ahoarau committed:
cmake: fallback to fetchcontent for nanobind

ahoarau committed:
README: add pixi section

ahoarau committed:
README: add pixi section

ahoarau committed:
cmake: downgrade nanobind to 2.9.2 to support python 3.8

ahoarau committed:
cmake: downgrade nanobind to 2.9.2 to support python 3.8

ahoarau committed:
cmake: change default options to mimic upstream

ahoarau committed:
cmake: change default options to mimic upstream

ahoarau committed:
cmake: fix property name

ahoarau committed:
cmake: fix property name

ahoarau committed:
cmake: use target_include_directories

ahoarau committed:
cmake: use target_include_directories

ahoarau committed:
ci: build in build folder

ahoarau committed:
ci: build in build folder

ahoarau committed:
cmake: add components test

ahoarau committed:
cmake: add components test

ahoarau committed:
ci: add components test

ahoarau committed:
ci: add components test

ahoarau committed:
pixi: add uninstall task

ahoarau committed:
pixi: add uninstall task

ahoarau committed:
cmake: add back LINK_PYTHON_INTERFACE_TO_OPENMP option

ahoarau committed:
cmake: add back LINK_PYTHON_INTERFACE_TO_OPENMP option

ahoarau committed:
ci: use multiline commands and cleanup

ahoarau committed:
ci: use multiline commands and cleanup

ahoarau committed:
cmake: fix link libs on python target

ahoarau committed:
cmake: fix link libs on python target

ahoarau committed:
ci: format

ahoarau committed:
ci: format

ahoarau committed:
cmake: format

ahoarau committed:
cmake: format

ahoarau committed:
cmake: the generator expression does not seem to work

ahoarau committed:
cmake: the generator expression does not seem to work

ahoarau committed:
ignore wheels

ahoarau committed:
ignore wheels

ahoarau committed:
pyproject: do not run tests when building wheels

ahoarau committed:
pyproject: do not run tests when building wheels

ahoarau committed:
ci: executables are in build dir

ahoarau committed:
ci: executables are in build dir

ahoarau committed:
cmake: limit to version >=3.22
probably the high limit will cause a CMAKE_CXX_COMPILER_CLANG_SCAN_DEPS-NOTFOUND error when using newer cmake versions

ahoarau committed:
cmake: limit to version >=3.22
probably the high limit will cause a CMAKE_CXX_COMPILER_CLANG_SCAN_DEPS-NOTFOUND error when using newer cmake versions

ahoarau committed:
cmake: disable testing by default

ahoarau committed:
cmake: disable testing by default

ahoarau committed:
ci: do not run executable

ahoarau committed:
ci: do not run executable

ahoarau committed:
ci: build with vectorization support for test to work

ahoarau committed:
ci: build with vectorization support for test to work

ahoarau committed:
ci: simplify conda script

ahoarau committed:
ci: simplify conda script

ahoarau committed:
ci: activate conda env according to the doc

ahoarau committed:
ci: activate conda env according to the doc

ahoarau committed:
ci: run on pwsh explicitely

ahoarau committed:
ci: run on pwsh explicitely

ahoarau committed:
ci: use CMD for windows conda

ahoarau committed:
ci: use CMD for windows conda

ahoarau committed:
ci: rename jobs to include compiler

ahoarau committed:
ci: rename jobs to include compiler

ahoarau committed:
c: add specific case for clang-cl

ahoarau committed:
c: add specific case for clang-cl

ahoarau committed:
ci: use cxx-compilers

ahoarau committed:
ci: use cxx-compilers

ahoarau committed:
ci: reset to shell: bash -el {0}

ahoarau committed:
ci: reset to shell: bash -el {0}

ahoarau committed:
ci: cxx-compiler (no s)

ahoarau committed:
ci: cxx-compiler (no s)

ahoarau committed:
ci: use CMAKE_C_COMPILER

ahoarau committed:
ci: use CMAKE_C_COMPILER

ahoarau committed:
ci: add msvc action

ahoarau committed:
ci: add msvc action

ahoarau committed:
ci: setup msvc first

ahoarau committed:
ci: setup msvc first

ahoarau committed:
ci: use cmd

ahoarau committed:
ci: use cmd

ahoarau committed:
ci: use CC and CXX

ahoarau committed:
ci: use CC and CXX

ahoarau committed:
ci: use local env vars

ahoarau committed:
ci: use local env vars

ahoarau committed:
ci: use set instead of env

ahoarau committed:
ci: use set instead of env

jorisv commented on .github/workflows/conda/environment.yml:
This can be removed. Pixi CI replace the conda-ci.

jorisv commented on .github/workflows/ci-linux-osx-win-conda.yml:
This should be removed. The Pixi CI will replace this workflow.

jorisv commented on bindings/python/proxsuite/__init__.py:
```diff
@@ -8,6 +8,12 @@
 if TYPE_CHECKING:
     from .proxsuite_pywrap import *  # noqa F403
 
+import os
+
+os.environ["KMP_DUPLICATE_LIB_OK"] = (
+    "True"  # to avoid issues with OpenMP in some environments
+)
+
```
Can you show me how to reproduce this issue ?

ahoarau replied:
It is simply ported from upstream https://github.com/Simple-Robotics/proxsuite/blob/devel/bindings/python/CMakeLists.txt#L254-L258

I have no way to verify if this is still useful/relevant.
Much better to put it here than the complicated code inside cmake with conditions. It should not hurt of not used.

jorisv replied:
Ok, this is done in this PR: https://github.com/Simple-Robotics/proxsuite/pull/240/changes
I think it's a pretty bad idea to setup this variable in our library, it should be done by users.

Since it's not due to JRL-v2, I will open an issue and I will try to check if it's still mandatory in another PR.

jorisv commented on examples/python/CMakeLists.txt:
```diff
@@ -1,11 +1,33 @@
-    "${PROJECT_NAME}-example-py-${EXAMPLE_NAME}"
-    "${EXAMPLE_FILE}"
-    "bindings/python"
+function(proxsuite_add_python_example name)
```
This should be added as a test

jorisv commented on examples/cpp/CMakeLists.txt:
```diff
@@ -1,20 +1,35 @@
-#
-# Copyright (c) 2022 INRIA
-#
+add_custom_target(proxsuite-examples-cpp)
 
-add_custom_target(${PROJECT_NAME}-example-cpp)
+function(proxsuite_add_example name)
+  add_executable(proxsuite-example-${name} ${name}.cpp)
+  target_link_libraries(proxsuite-example-${name} PRIVATE proxsuite)
+  add_dependencies(proxsuite-examples-cpp proxsuite-example-${name})
+endfunction()
```
This should be added as a test

jorisv commented on pixi.toml:
```diff
@@ -0,0 +1,123 @@
+  "Ninja",
+  "-DJRL_CMAKEMODULES_SOURCE_DIR=$JRL_CMAKEMODULES_SOURCE_DIR",
+  "--profiling-output=cmake-profiling.json",
+  "--profiling-format=google-trace",
```
To remove before merging

jorisv commented on CMakeLists.txt:
We must generate ament file before merging this PR

jorisv commented on CMakeLists.txt:
We must prefix all option with PROXSUITE_ when jrl_option support back compatibility option.

ahoarau replied:
Let's keep that for another PR. the cmake dependent options is not ported yet.

jorisv commented on CMakeLists.txt:
```diff
@@ -1,259 +1,329 @@
     FetchContent_Declare(
-      "jrl-cmakemodules"
-      GIT_REPOSITORY "https://github.com/jrl-umi3218/jrl-cmakemodules.git"
+      catch2
+      URL https://github.com/catchorg/Catch2/archive/refs/tags/v3.11.0.zip
+      URL_HASH MD5=ce96600812f04d19694f488ed2de75c5
+      EXCLUDE_FROM_ALL
     )
-    FetchContent_MakeAvailable("jrl-cmakemodules")
-    FetchContent_GetProperties("jrl-cmakemodules" SOURCE_DIR JRL_CMAKE_MODULES)
+    FetchContent_MakeAvailable(catch2)
+    include(${catch2_SOURCE_DIR}/extras/Catch.cmake)
```
Add a comment to know when we can remove this part (on the next Ubuntu LTS ?)

jorisv commented on CMakeLists.txt:
```diff
@@ -1,259 +1,329 @@
+# cereal (Serialization)
+if(BUILD_WITH_SERIALIZATION)
+  jrl_find_package(cereal CONFIG QUIET)
+  if(NOT cereal_FOUND)
+    message(
+      WARNING
+      "[proxsuite] cereal library not found. Falling back to FetchContent to get cereal v1.3.2"
+    )
+    include(FetchContent)
+    FetchContent_Declare(
+      cereal
+      URL https://github.com/USCiLab/cereal/archive/refs/tags/v1.3.2.zip
+      URL_HASH MD5=953a872cd3d78abf2e29212987a0ba71
+      SOURCE_SUBDIR
+      download_only
+      EXCLUDE_FROM_ALL
+    )
+    FetchContent_MakeAvailable(cereal)
+    add_library(cereal::cereal INTERFACE IMPORTED)
+    target_include_directories(cereal::cereal SYSTEM INTERFACE ${cereal_SOURCE_DIR}/include)
+  endif()
```
Add a comment to know when we can remove this part (on the next Ubuntu LTS ?)

jorisv commented on CMakeLists.txt:
```diff
@@ -1,259 +1,329 @@
+if(BUILD_PYTHON_INTERFACE)
+  jrl_find_python(3.8 REQUIRED COMPONENTS Interpreter Development.Module)
+  jrl_find_nanobind(2.4.0 CONFIG QUIET)
+  if(NOT nanobind_FOUND)
+    message(
+      WARNING
+      "[proxsuite] nanobind library not found. Falling back to FetchContent to get nanobind v2.9.2"
+    )
+    include(FetchContent)
+    FetchContent_Declare(
+      nanobind
+      GIT_REPOSITORY https://github.com/wjakob/nanobind.git
+      GIT_TAG v2.9.2
+      EXCLUDE_FROM_ALL
+    )
+    FetchContent_MakeAvailable(nanobind)
+  endif()
```
Add a comment to know when we can remove this part (on the next Ubuntu LTS ?)

jorisv reviewed (changes requested):
Nice work

ahoarau committed:
ci: temporarly disable nix ci

ahoarau committed:
ci: temporarly disable nix ci

ahoarau committed:
ci: uniformize names

ahoarau committed:
ci: uniformize names

ahoarau committed:
pixi.toml: remove cmake profiling

ahoarau committed:
pixi.toml: remove cmake profiling

ahoarau committed:
pixi.toml: add packaging tests

ahoarau committed:
pixi.toml: add packaging tests

ahoarau committed:
cmake: use local fetchcontent repo

ahoarau committed:
cmake: use local fetchcontent repo

ahoarau committed:
ci: remove conda workflow for pixi

ahoarau committed:
ci: remove conda workflow for pixi

ahoarau committed:
ci: integrate the conda workflow tests and variants

ahoarau committed:
ci: integrate the conda workflow tests and variants

ahoarau committed:
ci: merge configure and build to provide the cmake args

ahoarau committed:
ci: merge configure and build to provide the cmake args

ahoarau committed:
pixi.toml: pass build type and cxx standard via cmdline/env

ahoarau committed:
pixi.toml: pass build type and cxx standard via cmdline/env

ahoarau committed:
ci: use env to pass args to pixi

ahoarau committed:
ci: use env to pass args to pixi

ahoarau committed:
ci: rename to ci-ros

ahoarau committed:
ci: rename to ci-ros

ahoarau committed:
ci: enable additional build options

ahoarau committed:
ci: enable additional build options

ahoarau committed:
cmake: normalize PROXSUITE_FETCHCONTENT_SOURCE_DIR path for windows

ahoarau committed:
cmake: normalize PROXSUITE_FETCHCONTENT_SOURCE_DIR path for windows

ahoarau committed:
pixi.toml: use PROXSUITE_CXX_STANDARD instead

ahoarau committed:
pixi.toml: use PROXSUITE_CXX_STANDARD instead

ahoarau committed:
ci: use PROXSUITE_CXX_STANDARD instead

ahoarau committed:
ci: use PROXSUITE_CXX_STANDARD instead

ahoarau committed:
cmake: define PROXSUITE_CXX_STANDARD

ahoarau committed:
cmake: define PROXSUITE_CXX_STANDARD

ahoarau committed:
cmake: reformat with new gersemi version

ahoarau committed:
cmake: reformat with new gersemi version

ahoarau committed:
pre-commit: use new ruff-check id

ahoarau committed:
pre-commit: use new ruff-check id

ahoarau committed:
ci: make sure the env vars are available everywhere

ahoarau committed:
ci: make sure the env vars are available everywhere

ahoarau committed:
cmake: fix TO_CMAKE_PATH_LIST

ahoarau committed:
cmake: fix TO_CMAKE_PATH_LIST

ahoarau committed:
ci: build with --verbose

ahoarau committed:
ci: build with --verbose

ahoarau committed:
pixi.toml: remove CMAKE_VERBOSE_MAKEFILE
its meant to be used as arg (-D), not env. --verbose is better

ahoarau committed:
pixi.toml: remove CMAKE_VERBOSE_MAKEFILE
its meant to be used as arg (-D), not env. --verbose is better

ahoarau committed:
ci: build with only 2 jobs

ahoarau committed:
ci: build with only 2 jobs

ahoarau committed:
cmake: on windows, use fetchcontent to get catch2

ahoarau committed:
cmake: on windows, use fetchcontent to get catch2

ahoarau committed:
ci: remove the conda env

ahoarau committed:
ci: remove the conda env

ahoarau committed:
cmake: DOWNLOAD_EXTRACT_TIMESTAMP is not available on cmake 3.22

ahoarau committed:
cmake: DOWNLOAD_EXTRACT_TIMESTAMP is not available on cmake 3.22

ahoarau committed:
pixi.toml: relax deps contrains to allow to use cmake 3.22 exact

ahoarau committed:
pixi.toml: relax deps contrains to allow to use cmake 3.22 exact

ahoarau committed:
pixi.toml: add verbosity to tests

ahoarau committed:
pixi.toml: add verbosity to tests

ahoarau committed:
cmake: use only 3.22 to prevent the clang modules scanner to be triggerred
otherwise we need to add set_target_properties(run-proxqp PROPERTIES CXX_SCAN_FOR_MODULES OFF)

ahoarau committed:
cmake: use only 3.22 to prevent the clang modules scanner to be triggerred
otherwise we need to add set_target_properties(run-proxqp PROPERTIES CXX_SCAN_FOR_MODULES OFF)

ahoarau committed:
pixi.toml: add cpp20 and cpp23 envs

ahoarau committed:
pixi.toml: add cpp20 and cpp23 envs

ahoarau committed:
pixi.lock: update

ahoarau committed:
pixi.lock: update

ahoarau committed:
rename BENCHMARK to BENCHMARKS

ahoarau committed:
rename BENCHMARK to BENCHMARKS

ahoarau committed:
package.xml: drop ROS iron and python 2

ahoarau committed:
package.xml: drop ROS iron and python 2

ahoarau committed:
ci: update ROS CI and enable features

ahoarau committed:
ci: update ROS CI and enable features

ahoarau committed:
ci: disabling tests on windows+debug (too slow)

ahoarau committed:
ci: disabling tests on windows+debug (too slow)

ahoarau committed:
ci: disable fail-fast strategy

ahoarau committed:
package.xml: add python3-dev build depend

ahoarau committed:
package.xml: add python3-dev build depend

ahoarau committed:
python: import scipy.sparse.linalg explicitly for ubuntu 22

ahoarau committed:
python: import scipy.sparse.linalg explicitly for ubuntu 22

ahoarau committed:
ci: add build release and debug on ubuntu

ahoarau committed:
ci: add build release and debug on ubuntu

ahoarau committed:
ci: cosmetics

ahoarau committed:
ci: cosmetics

ahoarau committed:
ci: only build with 1 core to prevent swapping

ahoarau committed:
ci: only build with 1 core to prevent swapping

ahoarau committed:
rosci: add kilted and configure cache

ahoarau committed:
rosci: add kilted and configure cache

ahoarau committed:
implement uint128 on msvc

ahoarau committed:
implement uint128 on msvc

ahoarau committed:
fix julia test example

ahoarau committed:
fix julia test example

ahoarau committed:
gitignore: ignore dist/

ahoarau committed:
gitignore: ignore dist/

ahoarau committed:
cmake: remove extra cereal link

ahoarau committed:
cmake: remove extra cereal link

ahoarau committed:
cmake: make examples part of the test suite

ahoarau committed:
cmake: make examples part of the test suite

ahoarau committed:
cmake: add comments on fetchcontent fallbacks

ahoarau committed:
cmake: add comments on fetchcontent fallbacks

ahoarau committed:
python: import linalg explicitely for ubuntu 22

ahoarau committed:
python: import linalg explicitely for ubuntu 22

ahoarau committed:
cmake: add ros2 support

ahoarau committed:
cmake: add ros2 support

ahoarau committed:
ci: reorder the env vars

ahoarau committed:
ci: reorder the env vars

ahoarau committed:
cmake: generate ros2 python package files with condiition

ahoarau committed:
cmake: generate ros2 python package files with condiition

ahoarau committed:
ci: use latest master commit

ahoarau committed:
ci: use latest master commit

ahoarau committed:
pixi: numpy must be <2.4 due to nanobind limitation

ahoarau committed:
pixi: numpy must be <2.4 due to nanobind limitation

ahoarau committed:
cmake: check Eigen3::Eigen target before calling find_package

ahoarau committed:
cmake: check Eigen3::Eigen target before calling find_package

ahoarau committed:
cmake: use proxsuite::proxsuite target

ahoarau committed:
cmake: use proxsuite::proxsuite target

ahoarau committed:
cmake: use new option api

ahoarau committed:
cmake: use new option api

ahoarau committed:
gersemi: put back definitions

ahoarau committed:
gersemi: put back definitions

ahoarau committed:
cmake: reformat tests

ahoarau committed:
cmake: reformat tests

ahoarau committed:
cmake: add PROXSUITE_EXPORT_PACKAGE option

ahoarau committed:
cmake: add PROXSUITE_EXPORT_PACKAGE option

ahoarau committed:
cmake: add eigen fetchcontent support for testing

ahoarau committed:
cmake: add eigen fetchcontent support for testing

ahoarau committed:
cmake: fix new option API

ahoarau committed:
cmake: fix new option API

ahoarau committed:
pixi: isolate packaging tests

ahoarau committed:
pixi: isolate packaging tests

ahoarau committed:
pixi: update lock file

ahoarau committed:
pixi: update lock file

ahoarau committed:
pixi: update lock file

ahoarau committed:
pixi: update lock file

ahoarau committed:
cmake: only install the html folder, and putit in doc component

ahoarau committed:
cmake: only install the html folder, and putit in doc component

ahoarau committed:
cpp: make sure maros sparse test finds the helper header

ahoarau committed:
cpp: make sure maros sparse test finds the helper header

ahoarau committed:
cmake: add maros_meszaros label

ahoarau committed:
cmake: add maros_meszaros label

ahoarau committed:
cmake: remove the eigen3 find_package fallback

ahoarau committed:
cmake: remove the eigen3 find_package fallback

ahoarau committed:
pixi: refactor into features

ahoarau committed:
pixi: refactor into features

ahoarau committed:
ci: update with new pixi file

ahoarau committed:
ci: update with new pixi file

ahoarau committed:
pixi: update lock file

ahoarau committed:
pixi: update lock file

ahoarau committed:
rosci: do not use prerelease

ahoarau committed:
rosci: do not use prerelease

ahoarau committed:
cmake: update the get-jrl messages

ahoarau committed:
cmake: update the get-jrl messages

ahoarau committed:
pyproject.toml: add version constraints, use ninja, and format

ahoarau committed:
pyproject.toml: add version constraints, use ninja, and format

ahoarau committed:
pixi: build wheels

ahoarau committed:
pixi: build wheels

ahoarau committed:
pixi: update lock file

ahoarau committed:
pixi: update lock file

ahoarau committed:
ci: update wheel ci to use pixi

ahoarau committed:
ci: update wheel ci to use pixi

ahoarau committed:
ci: pypi only on release

ahoarau committed:
ci: pypi only on release

ahoarau committed:
cmake: require jrl >=1.2 to get the jrl-v2

ahoarau committed:
cmake: require jrl >=1.2 to get the jrl-v2

ahoarau committed:
cmake: update python and nanobind min required

ahoarau committed:
cmake: update python and nanobind min required

ahoarau committed:
pixi: update min python and nanobind version

ahoarau committed:
pixi: update min python and nanobind version

ahoarau committed:
pixi.toml: remove wheels and merge doc with default

ahoarau committed:
pixi.toml: remove wheels and merge doc with default

ahoarau committed:
pixi: update lock file

ahoarau committed:
pixi: update lock file

ahoarau committed:
ci: refactor with new pixi api

ahoarau committed:
ci: refactor with new pixi api

ahoarau committed:
pixi.toml: isolate doc feature

ahoarau committed:
pixi.toml: isolate doc feature

ahoarau committed:
pixi: update lock file

ahoarau committed:
pixi: update lock file

ahoarau committed:
pixi: add pixi build support

ahoarau committed:
pixi: add pixi build support

ahoarau committed:
gitignore: ignore conda files

ahoarau committed:
gitignore: ignore conda files

ahoarau committed:
pixi: fix pixi build tests

ahoarau committed:
pixi: fix pixi build tests

ahoarau committed:
pixi: update lock file

ahoarau committed:
pixi: update lock file

ahoarau committed:
ci: add pixi-build test

ahoarau committed:
ci: add pixi-build test

ahoarau committed:
ci: add explicit pixi env

ahoarau committed:
ci: add explicit pixi env

ahoarau committed:
ci: no need to install the env it's done in the setup

ahoarau committed:
ci: no need to install the env it's done in the setup

ahoarau committed:
cmake: require min macos SDK 10.14 (nanobind)
fixes the error on macos-15-intel:
error: aligned deallocation function of type 'void (void *, std::align_val_t) noexcep
   │ │ t' is only available on macOS 10.13 or newer

ahoarau committed:
cmake: require min macos SDK 10.14 (nanobind)
fixes the error on macos-15-intel:
error: aligned deallocation function of type 'void (void *, std::align_val_t) noexcep
   │ │ t' is only available on macOS 10.13 or newer

ahoarau committed:
ci: on pixi only keep 1 ubuntu and 1 windows

ahoarau committed:
ci: on pixi only keep 1 ubuntu and 1 windows

ahoarau committed:
cmake: include project version in config header generation

ahoarau committed:
cmake: include project version in config header generation

ahoarau committed:
ci: bump action versions

ahoarau committed:
ci: bump action versions

ahoarau committed:
pixi: uniformize constraints

ahoarau committed:
pixi: uniformize constraints

ahoarau committed:
pixi: comment out pixi-build

ahoarau committed:
pixi: comment out pixi-build

ahoarau committed:
pixi: update lock file
using eigen5

ahoarau committed:
pixi: update lock file
using eigen5

ahoarau committed:
cmake: disable AVX512 GEMM kernel for MSVC to avoid long compilation times with Eigen5

ahoarau committed:
cmake: disable AVX512 GEMM kernel for MSVC to avoid long compilation times with Eigen5

ahoarau committed:
pixi: update lock file

ahoarau committed:
pixi: update lock file

ahoarau committed:
uint128: constify and constexperify

ahoarau committed:
uint128: constify and constexperify

ahoarau committed:
tests: add more uint128_t tests

ahoarau committed:
tests: add more uint128_t tests

ahoarau committed:
ci: build tests on arch

ahoarau committed:
ci: build tests on arch

ahoarau committed:
tests: add shift operation tests for uint128_t with large shift amounts

ahoarau committed:
tests: add shift operation tests for uint128_t with large shift amounts

ahoarau committed:
uint128: use int32_t instead of int

ahoarau committed:
uint128: use int32_t instead of int

ahoarau committed:
uint128: remove comments

ahoarau committed:
uint128: remove comments

ahoarau committed:
tests: relax eps_abs for "test primal infeasibility solving"

ahoarau committed:
tests: relax eps_abs for "test primal infeasibility solving"

ahoarau committed:
cpp: update nothrow_destructible concept to use std::is_nothrow_destructible
fixes issue with vs generators

ahoarau committed:
cpp: update nothrow_destructible concept to use std::is_nothrow_destructible
fixes issue with vs generators

ahoarau committed:
cpp: explicit type to avoid std::max confusion on msvc

ahoarau committed:
cpp: explicit type to avoid std::max confusion on msvc

ahoarau committed:
gitignore: ignore vs folders

ahoarau committed:
gitignore: ignore vs folders

ahoarau committed:
uint128: only available on x64 builds

ahoarau committed:
uint128: only available on x64 builds

ahoarau committed:
pixi: update lock file

jorisv commented on .github/workflows/ci-linux.yml:
```diff
@@ -0,0 +1,90 @@
+        with:
+          path: ${{ env.CCACHE_DIR }}
+          key: ccache-${{ runner.os }}-${{ matrix.os }}-${{ github.sha }}
+          restore-keys: ccache-${{ runner.os }}-${{ matrix.os }}-
```
Replace the prefix `ccache` by an unique one.
Replace `runner.os` by `matrix.build_type`.

jorisv commented on .github/workflows/ci-linux.yml:
```diff
@@ -0,0 +1,90 @@
+    name: CI - ${{ matrix.os }} - ${{ matrix.build_type }} (APT)
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
```
Run on a docker image to ensure not using modified tools.

jorisv commented on .github/workflows/ci-pixi.yml:
```diff
@@ -0,0 +1,121 @@
+        with:
+          path: ${{ env.CCACHE_DIR }}
+          key: ccache-pixi-${{ matrix.os }}-${{ matrix.build_type }}-${{ matrix.cxx_std }}-${{ github.sha }}
+          restore-keys: ccache-pixi-${{ matrix.os }}-${{ matrix.build_type }}-${{ matrix.cxx_std }}-
```
Add `${{ matrix.pixi_environment }}` so we don't have cache issue if we add a new environment.

jorisv commented on .github/workflows/ci-pixi.yml:
```diff
@@ -0,0 +1,121 @@
+        run: pixi run -e default test-packaging
+
+      - name: Uninstall
+        run: pixi run -e default uninstall
```
Replace default by ${{ matrix.pixi_environment }}

jorisv commented on .github/workflows/ci-ros.yml:
```diff
@@ -0,0 +1,65 @@
+        with:
+          path: ${{ env.CCACHE_DIR }}
+          key: ccache-${{ matrix.ROS_DISTRO }}-${{github.run_id}}
+          restore-keys: ccache-${{ matrix.ROS_DISTRO }}-
```
```suggestion
          key: ccache-ci-ros-${{ matrix.ROS_DISTRO }}-${{github.run_id}}
          restore-keys: ccache-ci-ros-${{ matrix.ROS_DISTRO }}-
```

jorisv commented on cmake/get-jrl-cmakemodules.cmake:
```diff
@@ -0,0 +1,49 @@
+endif()
+
+# Fallback to FetchContent if not found
+set(JRL_GIT_REPOSITORY "https://github.com/ahoarau/jrl-cmakemodules.git")
```
To update when jrl-next is released

jorisv commented on include/proxsuite/proxqp/utils/uint128_msvc.hpp:
Is this file from you or taken from another project ?
In the last case, you need to cite the source.

ahoarau replied:
added copyright inria and note.

jorisv commented on test/src/dense_qp_wrapper.cpp:
```diff
@@ -3467,146 +3452,6 @@ TEST_CASE(
             << qp.results.info.solve_time << std::endl;
 }
 
-TEST_CASE(
-  "sparse random strongly convex qp with equality and "
-  "inequality constraints: test multiple solve at once with no initial guess")
```
Why this test is destroyed ? It was the same than another one ?

ahoarau replied:
I think it was removed because the name is the same as another test (catch2 does not allow that).

I'm putting back the test and renaming, good catch.

```
error: test case "sparse random strongly convex qp with equality and inequality constraints: test multiple solve at once with no initial guess", with tags "" already defined.
        First seen at proxsuite/test/cpp/dense_qp_wrapper.cpp:2184
        Redefined at proxsuite/test/cpp/dense_qp_wrapper.cpp:3457
```

jorisv commented on pixi.toml:
```diff
@@ -0,0 +1,296 @@
+# Activate color output with Ninja
+CMAKE_COLOR_DIAGNOSTICS = "ON"
+# Help ccache manage generated files and PCH (https://ccache.dev/manual/latest.html#_precompiled_headers)
+CCACHE_SLOPPINESS = "include_file_ctime,include_file_mtime,pch_defines,time_macros"
```
You need to setup an activation.{sh,bat} script like in others projects.

jorisv commented on pyproject.toml:
```diff
@@ -3,27 +3,31 @@ name = "proxsuite"
+  "-DBUILD_WITH_VECTORIZATION_SUPPORT:BOOL=ON",
+  "-DINSTALL_DOCUMENTATION:BOOL=OFF",
+  "-DBUILD_WITH_OPENMP_SUPPORT:BOOL=OFF",
+]
```
I don't think this part is still mandatory now the project build on python is managed by cmeel.

ahoarau replied:
will disapear after https://github.com/Simple-Robotics/proxsuite/pull/473 + rebase

jorisv commented on CMakeLists.txt:
```diff
@@ -1,259 +1,374 @@
+
+# cereal (Serialization)
+# NOTE: We have to keep the FetchContent support because:
+# - Cereal is not available via CMeel
```
It's fixed : https://pypi.org/project/cmeel-cereal/

jorisv commented on CMakeLists.txt:
```diff
@@ -1,259 +1,374 @@
+jrl_option(TEST_JULIA_INTERFACE "Run the julia examples as unittest" OFF)
+jrl_option(BUILD_WITH_OPENMP_SUPPORT "Build the library with the OpenMP support" OFF)
+jrl_option(LINK_PYTHON_INTERFACE_TO_OPENMP "Link OpenMP to the Python interface" ON CONDITION "BUILD_WITH_OPENMP_SUPPORT" FALLBACK OFF)
+jrl_option(BUILD_WITH_SERIALIZATION "Build with serialization support" ON CONDITION "BUILD_PYTHON_INTERFACE" FALLBACK OFF)
```
What's the link between serialization and python interface ?

ahoarau replied:
python binding requires BUILD_WITH_SERIALIZATION. Fixed 👍

jorisv commented on pixi.toml:
Take a look at other pixi.toml file in other maestro project.
We create different feature per optional dependency (here: simde, openmp, cereal) to allow developers to choose on which one they want to know.
Try to mimic as much as possible the one in pinocchio to be standard.

ahoarau replied:
updated. They are now uniform with the other projects. I'll let you figure out the extra env you want to add

jorisv commented on .github/workflows/ci-pixi.yml:
```diff
@@ -0,0 +1,121 @@
+    name: Prek CI
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v6
```
Please, use pre-commit and pre-commit.ci to be standard with other projects.

jorisv reviewed (changes requested):
When running tests, following tests are failing on Linux with pixi:

```bash
         33 - proxsuite-test-cpp.dense_maros_meszaros (SEGFAULT)
         34 - proxsuite-test-cpp.vectorized.dense_maros_meszaros (SEGFAULT)
         35 - proxsuite-test-cpp.sparse_maros_meszaros (SEGFAULT)
         36 - proxsuite-test-cpp.vectorized.sparse_maros_meszaros (SEGFAULT)
```

jorisv commented on CMakeLists.txt:
```diff
@@ -1,259 +1,374 @@
+jrl_option(BUILD_MAROS_MESZAROS_TESTS "Build the Maros-Meszaros tests, needs matio library" OFF)
+jrl_option(ENABLE_WARNINGS "Enable warnings during compilation" OFF)
+jrl_option(ENABLE_WARNINGS_AS_ERRORS "Treat all warnings as errors" OFF)
+jrl_option(BUILD_PYTHON_INTERFACE "Build the Python bindings" OFF)
```
This option is set to ON by default on jrl-v1

jorisv commented on examples/python/CMakeLists.txt:
```diff
@@ -1,11 +1,43 @@
+    )
+    set_tests_properties(
+      proxsuite-example-python.${name}
+      PROPERTIES ENVIRONMENT "PYTHONPATH=$<TARGET_FILE_DIR:proxsuite_pywrap>/.." LABELS python
```
It's better to use ENVIRONMENT_MODIFICATION to avoid issue with some package manager or custom installation path:
```
      ENVIRONMENT_MODIFICATION
        "PYTHONPATH=path_list_prepend:$<TARGET_FILE_DIR:proxsuite_pywrap>/.."
```

jorisv commented on test/python/CMakeLists.txt:
```diff
@@ -0,0 +1,25 @@
+  )
+  set_tests_properties(
+    proxsuite-test-python.${name}
+    PROPERTIES ENVIRONMENT "PYTHONPATH=$<TARGET_FILE_DIR:proxsuite_pywrap>/.." LABELS python
```
It's better to use ENVIRONMENT_MODIFICATION to avoid issue with some package manager or custom installation path:
```
      ENVIRONMENT_MODIFICATION
        "PYTHONPATH=path_list_prepend:$<TARGET_FILE_DIR:proxsuite_pywrap>/.."
```

ahoarau committed:
apply dos2unix

ahoarau committed:
pixi: update lock file

ahoarau committed:
replace prek by pre-commit

ahoarau committed:
ci-ros: bump commit

ahoarau committed:
ros-ci: add ros-ci prefix to cache key

ahoarau committed:
ci-pixi: replace default by ${{ matrix.pixi_environment }}

ahoarau committed:
ci-pixi: add ${{ matrix.pixi_environment }} to cache key to avoid future collisions

ahoarau committed:
ci-ubuntu: use docker images instead of official github images

ahoarau committed:
cmake: update jrl link to upstream

ahoarau committed:
cpp: add copyright inria

ahoarau committed:
tests: add back wrongly removed test and prevent name collision

ahoarau committed:
cmake: remove cereal fetchcontent
It is now supported on cmeel
https://pypi.org/project/cmeel-cereal/

ahoarau committed:
cmake: python bindings required BUILD_WITH_SERIALIZATION

ahoarau committed:
cmake: BUILD_PYTHON_INTERFACE defaults to ON
mimics jrl-v1

ahoarau committed:
cmake: use ENVIRONMENT_MODIFICATION

ahoarau committed:
pre-commit: update gersemi repo and version

ahoarau committed:
ci-ubuntu: add cereal and simde

ahoarau committed:
cmake: fix MAROS_MESZAROS_DIR dir path missing "/"

ahoarau committed:
pixi: uniformize pixi with other maestro projects

ahoarau committed:
ros-ci: add ros lyrical

ahoarau committed:
ci-pixi: add skip-deps

ahoarau committed:
pixi: add jrl-cmakemodules v2 dep

ahoarau committed:
cmake: add EXPECTED_TARGETS

ahoarau committed:
cmake: add warning and deprecated headers

ahoarau committed:
pixi: prefer kebab case

jorisv committed:
pixi: Fix doc environment build

jorisv committed:
pixi: Setup gitattributes for lockfile

jorisv commented on CMakeLists.txt:
```diff
@@ -1,263 +1,366 @@
+jrl_option(PROXSUITE_EXPORT_PACKAGE "Generate the CMake export files" ${PROJECT_IS_TOP_LEVEL})
+jrl_option(BUILD_TESTING "Build the tests" OFF)
+jrl_option(BUILD_EXAMPLES "Build the examples" OFF)
+jrl_option(BUILD_BENCHMARKS "Build the benchmarks" OFF)
```
```suggestion
jrl_option(BUILD_BENCHMARK "Build the benchmarks" OFF)
```

jorisv commented on CMakeLists.txt:
```diff
@@ -1,263 +1,366 @@
+
+jrl_option(PROXSUITE_EXPORT_PACKAGE "Generate the CMake export files" ${PROJECT_IS_TOP_LEVEL})
+jrl_option(BUILD_TESTING "Build the tests" OFF)
+jrl_option(BUILD_EXAMPLES "Build the examples" OFF)
```
```suggestion
jrl_option(PROXSUITE_BUILD_EXAMPLES "Build the examples" OFF)
```
BUILD_EXAMPLES is a new options

jorisv commented on CMakeLists.txt:
```diff
@@ -1,263 +1,366 @@
+jrl_option(BUILD_EXAMPLES "Build the examples" OFF)
+jrl_option(BUILD_BENCHMARKS "Build the benchmarks" OFF)
+jrl_option(BUILD_DOCUMENTATION "Build the documentation" OFF)
+jrl_option(BUILD_MAROS_MESZAROS_TESTS "Build the Maros-Meszaros tests, needs matio library" OFF)
```
```suggestion
jrl_option(PROXSUITE_BUILD_MAROS_MESZAROS_TESTS "Build the Maros-Meszaros tests, needs matio library" OFF)
```

BUILD_MAROS_MESZAROS_TESTS is a new option.

jorisv commented on CMakeLists.txt:
```diff
@@ -1,263 +1,366 @@
+jrl_option(BUILD_DOCUMENTATION "Build the documentation" OFF)
+jrl_option(BUILD_MAROS_MESZAROS_TESTS "Build the Maros-Meszaros tests, needs matio library" OFF)
+jrl_option(ENABLE_WARNINGS "Enable warnings during compilation" OFF)
+jrl_option(ENABLE_WARNINGS_AS_ERRORS "Treat all warnings as errors" OFF)
```
```suggestion
jrl_option(PROXSUITE_ENABLE_WARNINGS "Enable warnings during compilation" OFF)
jrl_option(PROXSUITE_ENABLE_WARNINGS_AS_ERRORS "Treat all warnings as errors" OFF)
```

`ENABLE_WARNINGS*` are new options.

`ENABLE_WARNINGS` activate `jrl_target_set_default_compile_options` on proxsuite traget. I think this should be done by default, why setting an option ?

ahoarau replied:
We don't want to export the warnings in the final package (it's an INTERFACE library). 
These options are only meant for proxsuite dev purposes.

jorisv commented on CMakeLists.txt:
```diff
@@ -1,263 +1,366 @@
+jrl_option(TEST_JULIA_INTERFACE "Run the julia examples as unittest" OFF)
+jrl_option(BUILD_WITH_OPENMP_SUPPORT "Build the library with the OpenMP support" OFF)
+jrl_option(LINK_PYTHON_INTERFACE_TO_OPENMP "Link OpenMP to the Python interface" ON CONDITION "BUILD_WITH_OPENMP_SUPPORT" FALLBACK OFF)
+jrl_option(BUILD_WITH_SERIALIZATION "Build with serialization support" ON)
```
```suggestion
jrl_option(PROXSUITE_BUILD_WITH_SERIALIZATION "Build with serialization support" ON)
```

New option

jorisv commented on CMakeLists.txt:
```diff
@@ -1,263 +1,366 @@
+jrl_option(LINK_PYTHON_INTERFACE_TO_OPENMP "Link OpenMP to the Python interface" ON CONDITION "BUILD_WITH_OPENMP_SUPPORT" FALLBACK OFF)
+jrl_option(BUILD_WITH_SERIALIZATION "Build with serialization support" ON)
+jrl_option(GENERATE_PYTHON_STUBS "Generate Python stubs" OFF)
+jrl_option(PYTHON_BINDING_CUSTOM_INSTALL_DIR "Override the default <current-python-install-sitelib>/<project-name>" "")
```
Never used ?

jorisv commented on CMakeLists.txt:
```diff
@@ -1,263 +1,366 @@
+jrl_option(INSTALL_DOCUMENTATION "Install the documentation" OFF)
+jrl_option(INITIALIZE_EIGEN_WITH_NAN "Initialize Eigen objects with NAN values" OFF)
+jrl_option(CHECK_RUNTIME_MALLOC "Check if some memory allocations are performed at runtime" OFF)
+jrl_option(SUFFIX_SO_VERSION "Suffix library name with its version" ON)
```
This option is never used.

jorisv commented on CMakeLists.txt:
```diff
@@ -1,263 +1,366 @@
+if(BUILD_TESTING)
+  if(WIN32)
+    set(Catch2_USE_FETCHCONTENT ON)
+  endif()
```
Can we add a guard to fetchcontent catch2 only on debug build configuration ? The CI will not have to rebuild catch 2 on release mode then.

jorisv commented on bindings/python/CMakeLists.txt:
```diff
@@ -1,313 +1,157 @@
+
+if(LINK_PYTHON_INTERFACE_TO_OPENMP)
+  message(STATUS "[proxsuite] Linking python bindings to OpenMP")
+  target_link_libraries(proxsuite_pywrap PRIVATE OpenMP::OpenMP_CXX)
```
Since proxsuite already link as INTERFACE OpenMP::OpenMP_CXX this is useless.
On old version, proxsuite target never link to OpenMP. It was manually appended to tests, binding and tests.

We could create a proxsuite-parallel and proxsuite-vectorized-parallel target that link to openmp ?
Or just remove proxsuite link to OpenMP::OpenMP_CXX, but I don't like it since it allow to use part of the library without proper linking.

ahoarau committed:
uint128: remove uused var

ahoarau committed:
update options: prefix with PROXSUITE for new, and rollback the old ones

jorisv committed:
python: Link all target to OpenMP when option is ON

jorisv commented on CMakeLists.txt:
`jrl_option(LINK_PYTHON_INTERFACE_TO_OPENMP "Link OpenMP to the Python interface" ON CONDITION "BUILD_WITH_OPENMP_SUPPORT" FALLBACK OFF)` doesn't work well.

If I run cmake, then activate BUILD_WITH_OPENMP_SUPPORT, LINK_PYTHON_INTERFACE_TO_OPENMP doesn't appear on cmake-gui/ccmake interface and stay at off. This create an invalid python module that doesn't link against OPENMP.

ahoarau replied:
You mean you first configure, and re-run cmake with `-DBUILD_WITH_OPENMP_SUPPORT=ON` ?
Can you run cmake with `--log-level=DEBUG` to see the output of jrl_option ?

jorisv replied:
To reproduce:

```bash
pixi shell
pixi run configure
cd build
cmake . -DBUILD_WITH_OPENMP_SUPPORT=ON -DGENERATE_PYTHON_STUBS=ON
```

Here the configuration summary:
```
================= Configuration Summary ==========================================================

Option                                   | Type     | Value | Description (default)
--------------------------------------------------------------------------------------------------
...
BUILD_PYTHON_INTERFACE                   | BOOL     | ON    | Build the Python bindings      (ON )
...
BUILD_WITH_OPENMP_SUPPORT                | BOOL     | ON    | Build the library with the Ope (OFF)
LINK_PYTHON_INTERFACE_TO_OPENMP          | BOOL     | OFF   | Link OpenMP to the Python inte (ON )
...
GENERATE_PYTHON_STUBS                    | BOOL     | ON    | Generate Python stubs          (OFF)
...
--------------------------------------------------------------------------------------------------
```

Then if I try to build the stubs:

```bash
ninja proxsuite_pywrap_stub
```

I get the following warning
```
...
/path/to/proxsuite-jrl-next/include/proxsuite/proxqp/parallel/qp_solve.hpp:33: warning: ignoring '#pragma omp parallel' [-Wunknown-pragmas]
   33 | #pragma omp parallel for schedule(dynamic)
...
```
And finally an import error when the stubgen try to import proxsuite:

```
ImportError: cannot import name 'instructionset' from partially initialized module 'proxsuite' (most likely due to a circular import) (/path/to/proxsuite-jrl-next/build/lib/site-packages/proxsuite/__init__.py)
```

The error message is not really clear but ldd show me the module is not linking against openmp but nm show it need some openmp symbols.

ahoarau replied:
fix available here: https://github.com/jrl-umi3218/jrl-cmakemodules/pull/867

jorisv committed:
doc: Fix installation and use an unique target name

nim65s commented on CMakeLists.txt:
```diff
@@ -1,263 +1,382 @@
-  ARCHIVE DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR}
-  RUNTIME DESTINATION ${CMAKE_INSTALL_FULL_BINDIR}
-)
+function(proxsuite_set_default_target_options target_name)
+  jrl_target_set_default_compile_options(${target_name} PRIVATE)
+  if(PROXSUITE_ENABLE_WARNINGS_AS_ERRORS)
+    jrl_target_treat_all_warnings_as_errors(${target_name} PRIVATE)
+  endif()
+endfunction()
```
This function seems to do pretty standard things. 
Could we add a `ENABLE_WARNINGS_AS_ERRORS` option to `jrl_target_set_default_compile_options` instead ?

nim65s commented on examples/python/CMakeLists.txt:
```diff
@@ -1,11 +1,46 @@
+      PROPERTIES
+        ENVIRONMENT_MODIFICATION
+          "PYTHONPATH=path_list_prepend:$<TARGET_FILE_DIR:proxsuite_pywrap>/.."
+        LABELS python
```
Nice, I did not know about these ctest labels, thanks !

ahoarau replied:
`pixi run test -L python` 🥇

nim65s commented on doc/CMakeLists.txt:
```diff
@@ -0,0 +1,72 @@
+if(INSTALL_DOCUMENTATION)
+  install(
+    DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/html
+    DESTINATION ${CMAKE_INSTALL_PREFIX}/share/doc/proxsuite
```
```suggestion
    DESTINATION ${CMAKE_INSTALL_DOCDIR}
```

ahoarau replied:
Cannot do that because of a current limitation in `jrl-cmakemodules` v2. Patch incomming

ahoarau replied:
fix by jrl upstream PR https://github.com/jrl-umi3218/jrl-cmakemodules/pull/857

nim65s reviewed (commented):
Very nice, thanks a lot !

I believe me miss the standalone bindings install option. I can add that later, but I'm mentionning that because for this setup it is better that we link stuff to the `proxsuite::xxx` alias instead of the direct `xxx`, because that option would add a conditional `jrl_find_package(proxsuite)` at the top. What would you think of this change ?

I'll push nix things soon

nim65s commented on doc/CMakeLists.txt:
```diff
@@ -0,0 +1,72 @@
+
+if(INSTALL_DOCUMENTATION)
+  install(
+    DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/html
```
```cmake
CMake Error at doc/cmake_install.cmake:54 (file):
  file INSTALL cannot find "/build/source/build/doc/html": No such file or
  directory.
```

I guess we need `set(DOXYGEN_HTML_OUTPUT html)` or something, because for me, that directory is currently called `doxygen-html`

jorisv replied:
It's weird, I don't have this issue when installing the documentation. It's generated in build/doc/html by default.

jorisv replied:
I use doxygen 1.9.1 and cmake 4.4.0

jorisv replied:
Anyway, I applied your suggestion

nim65s replied:
My bad, I had a `-DDOXYGEN_HTML_OUTPUT:STRING=doxygen-html` somewhere on the command line. That was probably caused by me trying to follow convention from https://github.com/jrl-umi3218/jrl-cmakemodules/blob/master/doxygen.cmake#L477

nim65s commented on CHANGELOG.md:
```diff
@@ -25,6 +25,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 - Upgraded nanobind dependency version (submodule) to v2.9.2 ([#418](https://github.com/Simple-Robotics/proxsuite/pull/418))
 - Better dynamic module handling ([#419](https://github.com/Simple-Robotics/proxsuite/pull/419))
 - Fix `SyntaxError` on importing `proxsuite.torch.qplayer` caused by invalid escape sequences in the `QPFunction` docstring ([#458](https://github.com/Simple-Robotics/proxsuite/pull/461))
+- Switch to [JRL CMake modules v2](https://github.com/jrl-umi3218/jrl-cmakemodules/pull/798) ([#28](https://github.com/Simple-Robotics/nanoeigenpy/pull/28))
```
it would probably be nice to document (here or in the doc) which CMake options were removed/added/renamed

jorisv replied:
We tried as much as possible to keep the build API. Main difference will be BUILD_DOCUMENTATION that will be set to OFF by default.

jorisv replied:
Actualy, it was already OFF by default on Proxsuite.

jorisv committed:
pixi: Create new-version environment

jorisv committed:
dev: Add release instruction

nim65s commented on development/release.md:
````diff
@@ -0,0 +1,28 @@
+```bash
+pixi shell -e new-version
+jrl-release --update-version X.Y.Z --git-commit --git-tag --sign-tag
+git push origin
+git push origin vX.Y.Z
+```
+
+Where `X.Y.Z` is the new version.
+Be careful to follow the [Semantic Versioning](https://semver.org/spec/v2.0.0.html) rules.
+
+Then, create a new release on [GitHub](https://github.com/Simple-Robotics/proxsuite/releases/new) with:
+
+* Tag: vX.Y.Z
+* Title: ProxSuite X.Y.Z
+* Body:
+```
+## What's Changed
+
+CHANGELOG CONTENT
+
+**Full Changelog**: https://github.com/Simple-Robotics/proxsuite/compare/vXX.YY.ZZ...vX.Y.Z
+```
+
+Where `XX.YY.ZZ` is the last release version.
````
This could be automatized with https://github.com/jrl-umi3218/jrl-cmakemodules/pull/852

jorisv committed:
doc: Set DOXYGEN_HTML_OUTPUT to html

ahoarau committed:
pixi: update doc target
